spring-context/src/test/java/org/springframework/context/groovy/GroovyBeanDefinitionReaderTests.java

```diff
@@ -0,0 +1,1299 @@
+
+			context.'component-scan'( 'base-package':"org.springframework.context.groovy" )
+		}
+		""").getBytes()));
+		appCtx.refresh();
+
+		var p = appCtx.getBean("person");
+		assertThat(p).isInstanceOf(AdvisedPerson.class);
+	}
+
+	@Test
+	void useSpringNamespaceAsMethod() {
+		var appCtx = new GenericApplicationContext();
+		var reader = new GroovyBeanDefinitionReader(appCtx);
+
+		reader.loadBeanDefinitions(new ByteArrayResource(("""
+		package org.springframework.context.groovy;
+		beans {
+			xmlns aop:"http://www.springframework.org/schema/aop"
+
+			fred(AdvisedPerson) {
+				name = "Fred"
+				age = 45
+			}
+			birthdayCardSenderAspect(BirthdayCardSender)
+
+			aop {
+				config("proxy-target-class":true) {
+					aspect( id:"sendBirthdayCard",ref:"birthdayCardSenderAspect" ) {
+						after method:"onBirthday", pointcut: "execution(void org.springframework.context.groovy.AdvisedPerson.birthday()) and this(person)"
+					}
+				}
+			}
+		}
+		""").getBytes()));
+
+		appCtx.refresh();
+
+		AdvisedPerson fred = (AdvisedPerson) appCtx.getBean("fred");
+		assertThat(fred).isInstanceOf(SpringProxy.class);
+		fred.birthday();
+
+		BirthdayCardSender birthDaySender = (BirthdayCardSender) appCtx.getBean("birthdayCardSenderAspect");
+
+		assertThat(birthDaySender.peopleSentCards).hasSize(1);
+		assertThat(birthDaySender.peopleSentCards.get(0).getName()).isEqualTo("Fred");
+	}
+
+	@SuppressWarnings("unchecked")
+	@Test
+	void useTwoSpringNamespaces() {
+		var appCtx = new GenericApplicationContext();
+		var reader = new GroovyBeanDefinitionReader(appCtx);
+		TestScope scope = new TestScope();
+		appCtx.getBeanFactory().registerScope("test", scope);
+
+		reader.loadBeanDefinitions(new ByteArrayResource(("""
+		package org.springframework.context.groovy
+		beans {
+			xmlns aop:"http://www.springframework.org/schema/aop"
+			xmlns util:"http://www.springframework.org/schema/util"
+			scopedList(Bean1) { bean ->
+				bean.scope = "test"
+				aop.'scoped-proxy'()
+			}
+			util.list(id: 'foo') {
+				value 'one'
+				value 'two'
+			}
+		}
+		""").getBytes()));
+		appCtx.refresh();
+
+		assertThat((List<String>)appCtx.getBean("foo")).contains("one", "two");
+
+		assertThat(appCtx.getBean("scopedList")).isNotNull();
+		assertThat(((Bean1)appCtx.getBean("scopedList")).getPerson()).isNull();
+		assertThat(((Bean1)appCtx.getBean("scopedList")).getPerson()).isNull();
+
+		// should only be true because bean not initialized until proxy called
+		assertThat(scope.instanceCount).isEqualTo(2);
+
+		appCtx = new GenericApplicationContext();
+		reader = new GroovyBeanDefinitionReader(appCtx);
+		appCtx.getBeanFactory().registerScope("test", scope);
+
+		reader.loadBeanDefinitions(new ByteArrayResource(("""
+		package org.springframework.context.groovy
+		import java.util.ArrayList
+		beans {
+			xmlns aop:"http://www.springframework.org/schema/aop",
+				util:"http://www.springframework.org/schema/util"
+			scopedList(Bean1) { bean ->
```

Comment:
```suggestion
			bean1(Bean1) { bean ->
```

Same as previous comment